comm/src/main/scala/coop/rchain/comm/unicast.scala

```diff
@@ -74,7 +79,10 @@ case class UnicastComm(local: PeerNode) extends Comm {
   override def recv: Try[Seq[Byte]] =
     try {
       receiver.receive(recv_dgram)
-      decode(recv_dgram.getData)
+      decode(recv_dgram.getData) match {
+        case Right(bytes) => Success(bytes)
+        case Left(err)    => Failure(err)
+      }
     } catch {
       case ex: Throwable => Failure(ex)
```

Comment:
Not sure that you really want to catch all throwables here, because it means that you also catch errors here (like OutOfMemory).
See https://github.com/alexandru/scala-best-practices/blob/master/sections/2-language-rules.md#28-must-not-catch-throwable-when-catching-exceptions

Reply:
I _think_ this catchall is the change you're requesting. Your recommendation is to bring the node down? I'm okay with that, I guess, but the reason I wanted to bubble it up is that this is the wrong level to try to handle something like OOM.

Reply:
Oh, I see I can use a capital-T `Try` block.